src/shared/apis/query.ts

```diff
@@ -92,6 +93,25 @@ export const processResponse = async (
   }
 }
 
+export const processResponseBlob = async (
```

Comment:
There's a bit of misdirection in this API, so it's not clear from this patch: does this mean we're using the `fetch` mechanism (or XHR family of apis) in the browser to do the download? It _appears_ that's so, but I want to be sure.

Reply:
Yes, we are.
We are specifically using all the same machineary used when running the normal (on submit) query, except that the post-processing (`processResponse`) is different for the csv download. That way we can:
* not allow any drift from the onSubmit query versus the onDownload query
* while still changing the post-request processing to not truncate for the onDownload.

Reply:
Agreed on the misdirection. This is because of how the existing APIs are structured. So instead of re-writing how the APIs are wired in this legacy part of the codebase, I choose to scope changes to providing a different post-network-call processor.

Reply:
I don't think this is what we want. We specifically _don't_ want to use the same machinery here. The browser already has capabilities for downloading this data, handling pauses and cancels, etc. The data here is of indeterminate size, and we don't want to be reading it into the browser.

Reply:
The "same machineary" refers to:
1. how we construct the query to send to the network call
2. the same `runQuery()` method, which every place outside of notebooks/qx-Builder uses to send the actual network call
3. how we notify the user if the network call fails.

I simply added another post-call-processor to the `runQuery()`, which could be optionally passed. This new post-call-processor does not read the data in the browser, but simply downloads it to a blob with a pointer reference.

Are you asking for us to create duplicate code for all 3 steps above, for only the csv download?

Reply:
Also...just as an fyi. The code `const blob = await response.blob()` takes the ReadableStream object (from the browser network call API response.body), and takes the pointed at stream and converts directly into a Blob/File object with a reference. The limits on this file object is the amount of memory available.

See the description here for why this was the best choice I could find, as I could not pipe directly from the Browser ReadableStream into a FileSystem writeable stream: https://github.com/influxdata/ui/pull/6036

Reply:
The only other way to completely bypass the browser is a change server-side, to update this response header for downloads:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

Every other frontend side approach is updating the post-processor.

Reply:
Yeah, I was absolutely expecting this to be the answer: we have to change the server side. We should look at what that takes. We had a implementation/feature here that was a partial download, presumably because we were downloading large files and blowing up the ram requirements of the browser. This patch would re-introduce that possible case.

Can we look at what it would take on the server side? I expect we'll have to make changes to Query™️.

Reply:
Made it work with direct filesystem download.